libraries/suid_check.rb

```diff
@@ -6,21 +6,44 @@ class SUIDCheck < Inspec.resource(1)
   name 'suid_check'
   desc 'Use the suid_check resource to verify the current SUID/SGID against a blacklist'
   example "
-     describe suid_check(blacklist) do
+     describe suid_check(blacklist, exclude_fstypes: %w[nfs nfs4], exclude_paths: %w[/proc]) do
        its('diff') { should be_empty }
      end
    "
 
-  def initialize(blacklist = nil)
-    blacklist = default if blacklist.nil?
+  def initialize(blacklist = nil, exclude_fstypes: nil, exclude_paths: nil)
     @blacklist = blacklist
+    @exclude_fstypes = exclude_fstypes || default_exclude_fstypes
+    @exclude_paths = exclude_paths || default_exclude_paths
+  end
+
+  def default_exclude_fstypes
+    %w[nfs nfs4 cifs smp nbd ceph]
+  end
+
+  def default_exclude_paths
+    %w[/sys /proc /var/lib/lxd/containers]
   end
 
   def permissions
-    output = inspec.command('find / -perm -4000 -o -perm -2000 -type f ! -path \'/proc/*\' ! -path \'/var/lib/lxd/containers/*\' -print 2>/dev/null | grep -v \'^find:\'')
+    exclude_opts = exclude_paths.map { |p| "-path #{p} -prune" }.join(' -o ')
+    output = inspec.command("find / \\( #{exclude_opts} \\) -o -perm -4000 -perm -2000 -type f -print 2>/dev/null " \
+                            "| grep -v '^find:'")
     output.stdout.split(/\r?\n/)
   end
 
+  def exclude_fstype_paths
+    mounts = inspec.command('mount')
```

Comment:
Should we open a ticket in inspec to expose their point parsing? https://github.com/inspec/inspec/blob/master/lib/resources/mount.rb

Reply:
It appears they're just inspecting files to see if they are mounts and getting their options from mount if so (https://github.com/inspec/inspec/blob/master/lib/resources/file.rb#L94), rather than parsing all mounts. I don't think the two are really compatible. Individual objects vs. "walking" mounts.